StatusMonitor/StatusMonitor.cs

```diff
@@ -90,56 +94,63 @@ public void start()
             {
                 Logging.Error("Directory " + Directory + " not supported: ", nsex);
             }
-            if (fileInfo.Exists)
+            if (fileInfo != null)
```

Comment:
This is a semantic change that I did not think you wanted. I think you want `(fileInfo?.Exists ?? false)`

Reply:
Fair point. I'll use `if (fileInfo != null && fileInfo.Exists)`

Reply:
Actually, let me amend that. We test for null here and we test whether the file exists further below.